lib/manager/components/transform/NormalizeField.js

```diff
@@ -0,0 +1,339 @@
+                <thead>
+                  <tr>
+                    <th>Substitute</th>
+                    <th>With</th>
+                    <th>Normalize space?</th>
+                    <th />
+                  </tr>
+                </thead>
+                <tbody>
+                  {substitutions && substitutions.map((s, i) => (
+                    <SubstitutionRow
+                      index={i}
+                      key={i}
+                      onChange={this._onChangeSubstitution}
+                      onRemove={this._onRemoveSubstitution}
+                      substitution={s}
+                    />
+                  ))}
+                  <tr>
+                    <td colSpan='4'>
+                      <Button bsSize='xsmall' onClick={this._onNewSubstitution}>
+                        <Icon type='plus' /> Add
+                      </Button>
+                    </td>
+                  </tr>
+                </tbody>
+              </Table>
+            </FormGroup>
+          </li>
+        </ul>
+        <div style={{marginBottom: '10px'}}>
+          <Button
+            bsSize='xsmall'
+            disabled={inputIsUnchanged}
+            onClick={this._onSaveTransformationSettings}
+            style={{marginRight: '5px'}}
+          >
+            Save
+          </Button>
+          <Button
+            bsSize='xsmall'
+            disabled={inputIsUnchanged}
+            onClick={this._onRevertTransformationSettings}
+            style={{marginRight: '5px'}}
+          >
+            Revert
+          </Button>
+        </div>
+      </div>
+    )
+  }
+}
+
+type GtfsFieldSelectorProps = {
+  onChange: (ReactSelectOption => void),
+  selectedField: ?string,
+  tableName: ?string
+}
+
+/**
+ * Obtains the desired GTFS spec table.
+ */
+function getGtfsTableSpec (tableName: string): ?GtfsSpecTable {
+  const spec = getGtfsSpec()
+  if (!spec) {
+    throw new Error('GTFS spec could not be found!')
+  }
+  const tableFile = `${tableName}.txt`
+  return spec.find(t => t.name === tableFile)
+}
+
+/**
+ * Dropdown selector to select a field from a given GTFS table.
+ */
+const GtfsFieldSelector = (props: GtfsFieldSelectorProps) => {
+  const { onChange, selectedField, tableName } = props
+  if (!tableName) return null
+
+  const table = getGtfsTableSpec(tableName)
+  if (!table) return null
+
+  return (
+    <Select
+      clearable={false}
+      onChange={onChange}
+      options={table.fields.map(field => ({label: field.name, value: field.name}))}
+      placeholder='Choose the field to normalize'
+      searchable={false}
+      style={{margin: '10px 0px', width: '230px'}}
+      value={selectedField}
+    />
+  )
+}
+
+type SubstitutionRowProps = {
+  index: number,
+  onChange: (Substitution, number) => void,
+  onRemove: number => void,
+  substitution: Substitution
+}
+
+/**
+ * Renders and lets user edit a substitution.
+ */
+class SubstitutionRow extends Component<SubstitutionRowProps> {
```

Comment:
Please place this component in a separate file. This file is too large.

Reply:
This was done in https://github.com/ibi-group/datatools-ui/pull/649/commits/7778d4bb3123d39936be338917e1331f1f93a89b.